vllm/worker/tt_worker.py

```diff
@@ -9,16 +9,164 @@
 from vllm.logger import init_logger
 from vllm.model_executor.layers.sampler import SamplerOutput
 from vllm.sequence import ExecuteModelRequest
-from vllm.utils import STR_DTYPE_TO_TORCH_DTYPE
+from vllm.utils import STR_DTYPE_TO_TORCH_DTYPE, get_dtype_size, is_pin_memory_available
+from vllm.worker.worker import raise_if_cache_size_invalid
 from vllm.worker.tt_model_runner import TTModelRunner
 from vllm.worker.worker_base import (LocalOrDistributedWorkerBase,
                                      LoraNotSupportedWorkerBase, WorkerInput)
 
 import ttnn
+from ttnn import ReplicateTensorToMesh
 
 logger = init_logger(__name__)
 
 
+class TTCacheEngine:
+    """Manages the KV cache.
+
+    This class is responsible for initializing and managing the TT and CPU KV
+    caches. It also provides methods for performing KV cache operations, such
+    as swapping and copying.
+    """
+
+    def __init__(
+        self,
+        cache_config: CacheConfig,
+        model_config: ModelConfig,
+        parallel_config: ParallelConfig,
+        device_config: DeviceConfig,
+    ) -> None:
+        self.cache_config = cache_config
+        self.model_config = model_config
+        self.parallel_config = parallel_config
+        self.device_config = device_config
+
+        self.head_size = model_config.get_head_size()
+        # Models like Jamba, have mixed typed layers, E.g Mamba
+        self.num_attention_layers = model_config.get_num_attention_layers(
+            parallel_config)
+        # TODO: should be 1 since TP=8
+        self.num_kv_heads = model_config.get_num_kv_heads(parallel_config)
+        # TODO: get num devices from device_config.device (device_mesh)
+        # TODO: add get_num_devices to worker
+        self.num_kv_heads //= 8 # TP=8, tries to use distributed worker if you give LLM 8 TP
+
+        self.block_size = cache_config.block_size
+        self.num_gpu_blocks = cache_config.num_gpu_blocks
+        if self.num_gpu_blocks:
+            self.num_gpu_blocks //= parallel_config.pipeline_parallel_size
+        self.num_cpu_blocks = cache_config.num_cpu_blocks
+        if self.num_cpu_blocks:
+            self.num_cpu_blocks //= parallel_config.pipeline_parallel_size
```

Comment:
We should remove all the usages of parallel_config here and in `_init_cache_engine` to simplify the code since we don't support the distributed features